### PR TITLE
Alik/feature/sharex data

### PR DESCRIPTION
**Pull request by tsyboom**

### CODE REVIEW

File: app/models/content.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+from tortoise import fields
+from tortoise.models import Model
+
+
+class Content(Model):
+    class Meta:
+        table = "contents"
+
+    id = fields.IntField(pk=True)
+    filename = fields.CharField(max_length=64)
```
Comment (kipod) [inline app/models/content.py]: max_length=260 (maximum file name length in windows)

File: app/models/content.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    content_type = fields.CharField(max_length=32)
```
Comment (kipod) [inline app/models/content.py]: max_length=64

File: app/models/user.py [hunk 3]
```diff
@@ -9,3 +9,4 @@ class Meta:
     id = fields.IntField(pk=True)
     username = fields.CharField(max_length=32)
```
Comment (kipod) [inline app/models/user.py]: ```max_length=64```
+
```python
email = fields.EMailField()
email_approved = fields.BooleanField()
```

File: app/models/user.py [hunk 4]
```diff
@@ -9,3 +9,4 @@ class Meta:
...
     hash_password = fields.CharField(max_length=128)
+    api_key = fields.CharField(max_length=128)
```
Comment (kipod) [inline app/models/user.py]: default value?

File: app/routers/auth.py [hunk 5]
```diff
@@ -9,16 +9,19 @@
...
 
 @router.post("/sign_in", response_model=Token)
 async def sign_in(form_data: OAuth2PasswordRequestForm = Depends()):
+    """Login"""
     service = AuthService()
     return await service.authenticate_user(form_data.username, form_data.password)
 
 
 @router.get("/user", response_model=User)
 def get_user(user: User = Depends(get_current_user)):
```
Comment (kipod) [inline app/routers/auth.py:26]: Does only Admin have access to this route?
If yes - add that into TODO: comment line

File: app/services/sharex.py [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+        log(
+            log.ERROR,
+            "Invalid request data, File[%s], Api Key[%s]",
+            check_file,
+            api_key,
+        )
+        raise exception("Could not validate form-data")
+    content_type = check_file.content_type
+    data = check_file.file.read()
+    filename = f"data/{str(uuid4())}"
```
Comment (kipod) [inline app/services/sharex.py]: better use absolute path (OK I know this is temp solution) Could you please add
```python
 # TODO: Store file on AWS S3
```

**Review by varvarashcherbyna7 — APPROVED** [review]

File: app/services/auth.py [hunk 7]
```diff
@@ -24,6 +25,7 @@ async def register_new_user(self, user_data: UserCreate) -> User:
         user = await UserDB.create(
             username=user_data.username,
             hash_password=self.hash_password(user_data.password),
+            api_key=str(uuid4()),
```
Comment (kipod) [inline app/services/auth.py]: IMHO this field can be filled by default value in DB Model

File: app/services/sharex.py [hunk 8]
```diff
@@ -0,0 +1,69 @@
+from uuid import uuid4
+
+from fastapi import Request, HTTPException, status
+
+from app.models import User, Content
+from app.logger import log
+
+
+def exception(msg: str):
```
Comment (kipod) [inline app/services/sharex.py]: raise_error(

File: tests/test_uploading_sharex.py [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+import asyncio
+from typing import Generator
+from fastapi.testclient import TestClient
+from tortoise.contrib.test import finalizer, initializer
+
+from app.setup import create_app
+from app.models import User, Content
+
+
+TEST_FILE = "/home/alik/simple2b/ShareX.portal/tests/test_data/Code_dPWe4YTTWI.png"
```
Comment (kipod) [inline tests/test_uploading_sharex.py]: this path on YOUR MACHINE!!!

File: tests/test_uploading_sharex.py [hunk 10]
```diff
@@ -0,0 +1,40 @@
+import pytest
```
Comment (kipod) [inline tests/test_uploading_sharex.py]: pytest - is not standard module ----> import after typing

File: tests/test_uploading_sharex.py [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+    data = {"api_key": response.json()["api_key"], "testing": True}
+    file = {"sharex": open(TEST_FILE, "rb")}
+    response = client.post("/sharex/", files=file, data=data)
+    assert response
+    assert response.ok
+    user = event_loop.run_until_complete(User.filter(api_key=data["api_key"]).first())
+    user_contents = event_loop.run_until_complete(user.contents)
+    assert user_contents
+    content = event_loop.run_until_complete(Content.filter(user=user).first())
+    assert content
```
Comment (kipod) [inline tests/test_uploading_sharex.py:43]: need to add here validation if the content stored correct

**Review by kipod — CHANGES_REQUESTED** [review]

File: tests/test_uploading_sharex.py (hunk 11 above)
Comment (tsyboom) [reply tests/test_uploading_sharex.py:43]: if data correct in record ?

Comment (kipod) [reply tests/test_uploading_sharex.py:43]: ну да :)

**Review by kipod — APPROVED** [review]

**Pull request merged**